iceoryx_posh/source/capro/service_description.cpp

```diff
@@ -120,11 +120,6 @@ ServiceDescription::ServiceDescription(const IdString_t& service,
 
 bool ServiceDescription::operator==(const ServiceDescription& rhs) const
 {
-    if (!isValid() || !rhs.isValid())
```

Comment:
What happens now if any of those is invalid? I may have missed it, maybe we do not even distinguish invalid from valid descriptions. Ideally there should be no invalid descriptions that can be created by the user.

Reply:
This was changed due to the fact that the `ServiceRegistry` can now store invalid `ServiceDescriptions` and we will check for validity on a higher-level closer to the user API. The changes allows the `ServiceRegistry` to match and remove two invalid `ServiceDescription`s. The topic on how to prevent the user from creating invalid `ServiceDescription` was heavily discussed in #854 #852 and #853 so far we did not come up with a nice user-friendly solution. It's not super user-unfriendly if we stick with the current approach IMHO.